Boolean.CSharp.Main/Account.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Principal;
+using System.Threading.Tasks;
+
+namespace Boolean.CSharp.Main
+{
+
+    public enum AccountType {
+        CURRENT = 1,
+        SAVINGS = 2,
+    }
+    public abstract class Account : IAccount
+    {
+        public string AccountName { get; private set;}
+        public int ID { get; private set; }
+        public Branch AccountBranch { get; private set; }
+        private List<Transactions> _transactions;
+
+        public Account(string accountname, decimal balance, Branch accosiatedBranch) {
+            ID = GetRandomID();
+            _transactions = new List<Transactions>();
+            if (!string.IsNullOrEmpty(accountname)) {
+                AccountName = accountname;
+            } else {
+                throw new Exception("You can not have an empty account name");
+            }
+            
+
+            if (balance > 0) {
+                AddTransaction(balance, TransactionType.DEPOSIT);
+            } else {
+                throw new Exception($"{balance} is not a valid starting balence, the balance must be 0 or greater");
+            }
+
+            if (!string.IsNullOrEmpty(accosiatedBranch.ToString())) {
+                AccountBranch = accosiatedBranch;
+            } else {
+                throw new Exception("No entered branch");
+            }
+
+            
+        }
+
+
+        // Extension
+        private int GetRandomID() {
```

Comment:
random ids are not good because what if the random number generator creates an Id that is already in use? better to store an `int lastUsedId` in this class and increment it each time you create a transaction.